### PR TITLE
[SYCL][DOC] Add wording about buffer host data copies

### PR DESCRIPTION
**Pull request by Bensuo**
- Adds wording about buffers taking host data copies in some situations
- Introduces a new graph property which disables copy behaviour
- Add property list to begin_recording()

### CODE REVIEW

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc [hunk 1]
```diff
@@ -321,6 +321,14 @@ enum class graph_state {
   executable
 };
 
+namespace property{
```
Comment (EwanC) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]:
```suggestion
namespace property { 
```

alternatively, could put this in the property namespace above we have for `property::node::depends_on`

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc [hunk 2]
```diff
@@ -1010,7 +1031,7 @@ be immediately stale, any code which relies on queue waits should take care
 to ensure waits are not performed on queues in recording mode. For example, by
 using separate queues for graph recording and normal queue operations.
 
-=== Storage Lifetimes
+=== Storage Lifetimes [[storage-lifetimes]]
```
Comment (EwanC) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc:1046]: shouldn't need the extra `[[storage-lifetimes]]` in order to reference earler with `<<storage-lifetimes, storage lifetimes>>`

**Review by EwanC — COMMENTED** [review]
minor nits

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc (hunk 2 above)
Comment (Bensuo) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc:1046]: This is true, though it does prevent the section ID from changing if the section title is modified, or if properties which modify the way the ids for the document are generated are modified.

I can remove it if need be though!

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc (hunk 1 above)
Comment (Bensuo) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: I think I prefer keeping the properties near their associated classes for readability, if possible!

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc (hunk 2 above)
Comment (EwanC) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc:1046]: That makes sense from an editorial maintenance point-of-view

**Review by julianmi — APPROVED** [review]
LGTM

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc [hunk 3]
```diff
@@ -321,6 +321,14 @@ enum class graph_state {
   executable
 };
 
+namespace property { 
+namespace graph {
+
+class no_host_copy;
```
Comment (reble) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]:
```suggestion
class no_host_copy {
public:
  no_host_copy() = default;
};
```

**Review by reble — APPROVED** [review]

**Review by EwanC — APPROVED** [review]

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc [hunk 4]
```diff
@@ -333,8 +341,8 @@ public:
   command_graph<graph_state::executable>
   finalize(const context& syclContext, const property_list& propList = {}) const;
 
-  bool begin_recording(queue& recordingQueue);
-  bool begin_recording(const std::vector<queue>& recordingQueues);
+  bool begin_recording(queue& recordingQueue, const property_list& propList = {});
+  bool begin_recording(const std::vector<queue>& recordingQueues, const property_list& propList = {});
```
Comment (EwanC) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc:348]: Realised the signatures in the table of queue recording member functions will need updated too

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc (hunk 3 above)
Comment (Bensuo) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: Done!

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc (hunk 4 above)
Comment (Bensuo) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc:348]: Good point, I've added them now!

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc [hunk 5]
```diff
@@ -666,7 +695,8 @@ Exceptions:
 |
 [source, c++]
 ----
-bool begin_recording(const std::vector<queue>& recordingQueues)
+bool begin_recording(const std::vector<queue>& recordingQueues, 
+const property_list& propList = {})
```
Comment (EwanC) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: I should at least be consistent across PRs with my nitpicking

```suggestion
bool begin_recording(const std::vector<queue>& recordingQueues, 
                     const property_list& propList = {})
```

Comment (Bensuo) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: Fixed!

**Review by EwanC — APPROVED** [review]

**Pull request merged**